lib/src/Base/Stat/UserDefinedStationaryCovarianceModel.cxx

```diff
@@ -69,6 +69,29 @@ UserDefinedStationaryCovarianceModel::UserDefinedStationaryCovarianceModel(const
   }
 }
 
+/* Easy constructor for 1D outputs*/
+UserDefinedStationaryCovarianceModel::UserDefinedStationaryCovarianceModel(const RegularGrid & mesh,
+    const Collection<Scalar> & covarianceValues)
+  : StationaryCovarianceModel()
+  , covarianceCollection_(0)
+  , mesh_(mesh)
+  , nearestNeighbour_(mesh)
+{
+  const UnsignedInteger size = mesh.getVerticesNumber();
+  if (size != covarianceValues.getSize())
+    throw InvalidArgumentException(HERE) << "Error: a covariance value should correspond to each distance specified in the regular grid. "
+                                         << "Regular grid size = " << size << " vs number of covariance values in the collection = " << covarianceValues.getSize();
+  covarianceCollection_ = CovarianceMatrixCollection(size);
```

Comment:
Replace it by:
```
covarianceCollection_ = CovarianceMatrixCollection(size, CovarianceMatrix(1));
 for (UnsignedInteger k = 0; k < size; ++k)
  {
    if (covarianceValues[k] < 0)
      throw InvalidArgumentException(HERE) << k << "-th covariance value is " << covarianceValues[k] << ", should be nonnegative.";
    covarianceCollection_[k](0, 0) = currentCovarianceMatrix;
  }
```

Reply:
I have adapted your code (`currentCovarianceMatrix` is not needed).